test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -2027,6 +2027,44 @@ def test_cgroup_cpuset_memory_spread_page(self):
         self.assertEqual(result['rc'], 0)
         self.assertEqual(result['out'][0], '1')
 
+    def test_cgroup_use_hierarchy(self):
+        """
+        Test that memory.use_hierarchy is enabled by default
+        when PBS cgroups hook is instantiated
+        """
+
+        now = int(time.time())
+
+        # Remove PBS directories from memory subsystem
+        if 'memory' in self.paths and self.paths['memory']:
+            cdir = self.paths['memory']
+            if os.path.isdir(cdir):
+                cpath = os.path.join(cdir, 'pbspro')
+                if not os.path.isdir(cpath):
+                    cpath = os.path.join(cdir, 'pbspro.slice')
+        cmd = ["rmdir", cpath]
+        self.logger.info("Removing %s" % cpath)
+        self.du.run_cmd(cmd=cmd, sudo=True)
+
+        self.load_config(self.cfg6 % (self.swapctl))
+
+        self.momA.restart()
+
+        # Wait for exechost_startup hook to run
+        self.momA.log_match("Hook handler returned success for"
+                            " exechost_startup event",
+                            starttime=now)
+
+        # Verify that memory.use_hierarchy is enabled
+        fpath = os.path.join(cpath, "memory.use_hierarchy")
+        self.logger.info("looking for file %s" % fpath)
+        if os.path.isfile(fpath):
+            file = open(fpath, "r")
```

Comment:
Please open the file within a "with" statement to ensure it gets closed. Also, as Codacy indicated, don't use a variable name that conflicts with Python builtin methods. The cgroup hook commonly uses "desc" to store the file descriptor returned from open.